add sbom artifacts to model

houdini91 opened the pull request:
Hi,
I noticed using the sbom cataloger the relations that are read from the sbom are not aggregated in to the Syft model sbom.

I have a small fix suggestion, not sure if there was a reason behind it or simply a typo.

Are there any tests you would like me to add for this PR?

Signed-off-by: houdini91 <mdstrauss91@gmail.com>

kzantow commented:
@houdini91 this looks good but are you able to fix the tests?

kzantow commented on syft/pkg/cataloger/internal/pkgtest/test_generic_parser.go:
```diff
@@ -179,9 +180,29 @@ func (p *CatalogTester) assertPkgs(t *testing.T, pkgs []pkg.Package, relationshi
 
-	if diff := cmp.Diff(p.expectedRelationships, relationships, p.compareOptions...); diff != "" {
-		t.Errorf("unexpected relationships from parsing (-expected +actual)\n%s", diff)
+	if err := SubsetRelationship(p.expectedRelationships, relationships, p.compareOptions...); err != nil {
+		diff := cmp.Diff(p.expectedRelationships, relationships, p.compareOptions...)
+		t.Errorf("unexpected %s relationships from parsing (-expected +actual)\n%s", err, diff)
```
I'm not entirely sure this is ideal -- if there is a diff, the diff will be very noisy and not represent what the check is actually doing. Could we just update the individual tests to have the right expected relationships?

houdini91 replied:
Np, do you want me to as simply 150 relationship to the static go code? Should we remove some from the sbom?

houdini91 replied:
Fyi the diff line your referring is simply to output for failed subset check above it was simpler for me to use it's full diff output for test outpu
But I understand you concern about the subset comparison

kzantow replied:
I'd be good with updating the source SBOM to make a manageable set of relationships to add explicitly to the tests.

kzantow commented:
@houdini91 I think you may also just need to run `make lint-fix` for the static analysis failure

houdini91 commented:
@kzantow Hi, sorry for the delay, i admittedly forgot about this PR.
Fixed your suggestions, rebased and squashed (i forgot to sign some commit).
Have a look seem like the tests pass.

kzantow reviewed (approved):
LGTM 👍

kzantow commented:
Thanks @houdini91 !
